src/style-spec/expression/index.js

```diff
@@ -135,73 +100,86 @@ function createExpression(expression: mixed,
         };
     }
 
-    const isFeatureConstant = isConstant.isFeatureConstant(parsed);
-    if (!isFeatureConstant && context === 'property' && !propertySpec['property-function']) {
-        return {
-            result: 'error',
-            errors: [new ParsingError('', 'property expressions not supported')]
-        };
+    return success({ evaluate, parsed });
+}
+
+type ConstantExpression = {
+    kind: 'constant',
+    evaluate: (globals: GlobalProperties, feature?: Feature) => any
```

Comment:
It's not feasible yet, but I hope to be able to specialize the `evaluate` method appropriately for the various kinds in a later commit. For example, constant expression evaluation should require no arguments, and the `Feature` argument should not be options for source and composite expression evaluation.

Reply:
Once this happens, we'll need a way to heatmap-color.